app/models/project.rb

```diff
@@ -67,4 +66,20 @@ def clone_projects_into(clone)
       sub_project.clone_stories_into(sub_project_clone)
     end
   end
+
+  private
+
+  def archive
+    update_column :status, "archived"
+    projects.each do |sub_project|
+      sub_project.update_column :status, "archived"
+    end
+  end
+
+  def unarchive
+    update_column :status, nil
+    projects.each do |sub_project|
```

Comment:
🥦 Wonder if something like this is a good idea?
```suggestion
    (self + projects).each do |sub_project|
```

Reply:
maybe something like `Project.where(parent_id: id).update_all(status: nil)` is more performant, we don't really need to load each project to update them

or even `Project.where(id: id).or(Project.where(parent_id: id)).update_all(status: nil)` and we can update the parent and sub-projects all in one query

Reply:
Thank you, this is a great suggestion!